### DISCUSSION TITLE
summerizer

### DISCUSSION THREAD

**Discussion by DavidMacha** [openai/openai-python#836] [Q&A] [answered]
Hello everyone! I'm encountering an issue with the OpenAI Python library. I've integrated it into my text summarization project, but the generated summaries are not meeting my expectations. They are either too lengthy or not capturing the crucial information from the input articles. Can someone help me troubleshoot this issue?

```
import openai

# Set my API key
api_key = 'my_api_key_here'
openai.api_key = api_key

# Provide an article for GPT-3 to summarize
article = """//(long article here) ..."""

# Make an API call to generate a summary
response = openai.Completion.create(
    engine='text-davinci-002',
    prompt=article,
    max_tokens=150  # I've experimented with token count, but the summaries are still not ideal
)

# Print the generated summary
print(response.choices[0].text)

```

**Comment by Mattral** [comment, marked as answer]
Your code looks fine. One thing to consider is the specificity of your prompt. GPT models can be sensitive to the wording. Instead of a general prompt, you might try being more explicit, like "Summarize the key points from the following article."
 have you experimented with the temperature parameter in your API call? It can influence the randomness of the generated text. Lower values like 0.2 might provide more focused and deterministic summaries.

Additionally,  you might want to experiment with the max_tokens parameter more. Adjusting it to a smaller value could help in controlling the length of the generated summaries.

**Reply by DavidMacha** [reply]
Thanks alots Thats really helpful
